### DISCUSSION TITLE
fetching/searching records that don't exist throws an error

### DISCUSSION THREAD

**Discussion by MobinAskari** [pocketbase/pocketbase#3988] [Q&A]
I read the docs and can't make sense of it, which I understand is probably my fault. But I only want the searching requests to throw an error if the database is down , the request is unauthorized or something else in this ballpark. Throwing an error just because the record doesn't exist seems unintuitive and makes it harder to achieve the desired behavior (at least in my use case). 

So basically, I don't want the JS SDK to throw just because the record doesn't exist, and instead I need it to return normally, maybe with data being `null`, for example

**Comment by jonshipman** [comment]
My usual workflow is to wrap API requests with the frontend framework. So in practice that means only performing pocketbase connections on the server. By doing so, you can wrap those in a try...catch and only throw the errors you want.

Take a look at my boilerplate setup here: https://github.com/jonshipman/sveltekit-pb-boilerplate

Specifically, the https://github.com/jonshipman/sveltekit-pb-boilerplate/blob/main/%5Bweb%5D/src/routes/api/%5Bcollection%5D/%2Bserver.ts file. In this example, I'm avoiding leaking the `isAbort` error up to the client.

**Reply by MobinAskari** [reply]
Thank for taking the time to answer but I'm not using Pb in a frontend environment. I have a Fastify server that has a /getPost endpoint which first checks for the post in the Db, and if non exists, it fetches the post over the network from another API.

This current/default behavior of Pb SDK throwing an error just because the record doesn't exist makes for really ugly code/checks just to achieve the desired result.

An example of this:
```typescript
let dbGetResult;
    try {
      const alreadySavedPost = await getPostFromDb(tweetLink);

      dbGetResult = alreadySavedPost;

    } catch (error: any) {
      dbGetResult = error;
    }

    if ("status" in dbGetResult && dbGetResult.status !== 404)
      return new errorResponse(dbGetResult.status || 0, dbGetResult.message);

    if ("postURL" in dbGetResult) 
      return new successResponse(200, dbGetResult);
```

The getPostFromDb is a wrapper around the getFirstListItem and I could perform a trycatch there, but I would lost the benefits of composable errors
